fix(x11/python-torch): include numpy files when building (fixes #21744)

AminurAlam opened the pull request:
see https://github.com/termux/termux-packages/issues/21744

fornwall reviewed (changes requested):
This doesn't work - NumPy support is still not built, check the build log:

> 2024-10-10T14:13:30.6174431Z   NumPy could not be found.  Not building with NumPy.  Suppress this warning
> 2024-10-10T14:13:30.6175343Z   with -DUSE_NUMPY=OFF

We should probably patch away this fallback, to instead fail the build if not building the desired numpy support - https://github.com/pytorch/pytorch/blob/main/cmake/Dependencies.cmake#L886-L892:

```cmake
    if(USE_NUMPY)
      if(NOT Python_NumPy_FOUND)
        message(WARNING "NumPy could not be found. Not building with NumPy. Suppress this warning with -DUSE_NUMPY=OFF")
        caffe2_update_option(USE_NUMPY OFF)
      else()
        caffe2_update_option(USE_NUMPY ON)
      endif()
```

fornwall commented:
Relevant change in pytorch regarding the problem: https://github.com/pytorch/pytorch/commit/8ea1dc874880de9bbcf32699c4098a3c5e3f3b20, `Use Python::NumPy target`.

licy183 commented on x11-packages/python-torch/build.sh:
```diff
@@ -20,7 +20,7 @@ TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
 -DCMAKE_INSTALL_PREFIX=${TERMUX_PKG_SRCDIR}/torch
 -DCMAKE_PREFIX_PATH=${TERMUX_PYTHON_HOME}/site-packages
--DNumPy_INCLUDE_DIR=${TERMUX_PYTHON_HOME}/site-packages/numpy/_core/include
+-DNUMPY_INCLUDE_DIR=${TERMUX_PYTHON_HOME}/site-packages/numpy/_core/include
```
```suggestion
-DPython_NumPy_INCLUDE_DIR=${TERMUX_PYTHON_HOME}/site-packages/numpy/_core/include
```

AminurAlam commented:
i have added a patch for fail and renamed the flag
```diff
-        message(WARNING "NumPy could not be found. Not building with NumPy. Suppress this warning with -DUSE_NUMPY=OFF")
+        message(FATAL_ERROR "NumPy could not be found. Not building with NumPy. Suppress this warning with -DUSE_NUMPY=OFF")
```

fornwall commented:
Nice work, seems to work beautifully now!
